reference/pipeline/expressions/index.md

```diff
@@ -0,0 +1,214 @@
+`${ }`. Within these brackets, you can add arbitrary expressions to access and
+modify the details of your pipeline during pipeline execution.
+
+### Code
+
+Spinnaker allows you to execute Java code within a pipeline expression. This can
+be useful for string manipulation or to build more complex expressions than
+would otherwise be possible. For security reasons, you can only use whitelisted
+Java classes. You can find the full list of whitelisted classes
+[here](#whitelisted-java-classes).
+
+You can use methods available to the whitelisted classes without importing them.
+For example, you can process a list of values entered as a parameter using
+Java's String `split()` function, provided users enter them using a standard
+split character. You can process the list `us-east-1,us-west-1,eu-west-1` with
+the expression `${parameters.regions.split(",")}`.
+
+You can instantiate new classes inside of an expression using the fully
+qualified package name. For example, you might want to use the [SimpleDateFormat](https://docs.oracle.com/javase/8/docs/api/java/text/SimpleDateFormat.html)
+class to get the current date in MM-DD-YYYY format. You can do this using the
+expression `${new java.text.SimpleDateFormat("MM-DD-YYYY").format(new
+java.util.Date())}`.
+
+Similarly, you can call static methods using the syntax
+`T(fully.qualified.class.name).methodName()`. For example, to calculate the date
+5 days from now, you can call:
+`${T(java.time.LocalDate).now().plusDays(5).toString()}`.
+
+### Comparisons
+
+You can use relational operators to compare values in an expression, such as
+`${instance["size"] > 400}` or `${parameters["runCanary"] == "true"}`.
+
+Note that you may need to transform some values for the comparisons to work
+properly. In the example above, `parameters["runCanary"]` returns a string
+rather than a boolean. To use it in a comparision, you need to either compare
+it to a string or convert it to a boolean:
+`${#toBoolean(parameters["runCanary"]) == true}`.
+
+Another example is that the status attribute of a stage is actually an enum
+internally, not a string. To compare the status to a string, you need to call
+`.toString()` on the result. For example:
+`${#stage("Deploy")["status"].toString() == "SUCCEEDED"}`.
+
+### Functions
+
+The expression language has several built in helper functions that simplify
+common use cases. The syntax for calling built-in functions is
+`#functionName(params)`. For example, `#fromUrl("http://www.netflix.com")`. See
+the full list of functions [here](#helper-functions).
+
+### Lists
+
+You can also use lists in your expressions. Spinnaker always provides a list
+called _stages_, which you can use to access each stage by index:
+`${execution["stages"][0]}` returns the value of the first stage in your
+pipeline.  Keep in mind, though, that if you change the order in which your
+stages execute in the pipeline (separately from your expression), the value of
+`stages[i]` will also change.
```

Comment:
Worth noting that this is for illustration purposes only and the less fragile approach is to use the #stages( 'stagename' ) helper